pkg/nsx/services/subnetport/builder.go

```diff
@@ -39,16 +59,34 @@ func (service *SubnetPortService) buildSubnetPort(obj interface{}, nsxSubnet *mo
 	}
 	var externalAddressBinding *model.ExternalAddressBinding
 	var addressBindings []model.PortAddressBindingEntry
+	var isIPPool bool
 	switch o := obj.(type) {
 	case *v1alpha1.SubnetPort:
 		externalAddressBinding = service.buildExternalAddressBinding(o)
-		if restoreMode && o.Status.NetworkInterfaceConfig.IPAddresses[0].IPAddress != "" {
-			ip := strings.Split(o.Status.NetworkInterfaceConfig.IPAddresses[0].IPAddress, "/")[0]
-			addressBindings = []model.PortAddressBindingEntry{
-				{
-					IpAddress:  &ip,
-					MacAddress: &o.Status.NetworkInterfaceConfig.MACAddress,
-				},
+		// NSX only supports one IP per SubnetPort for now, but we implement for the common case
+		// supporting multi-IP as the API accepts the list
+		if len(o.Spec.AddressBindings) > 0 {
+			for _, addressBinding := range o.Spec.AddressBindings {
+				addressBindings = append(addressBindings, model.PortAddressBindingEntry{
+					IpAddress:  &addressBinding.IPAddress,
+					MacAddress: &addressBinding.MACAddress,
+				})
+				// Chck if the specified MAC is in NSX MAC Pool to determine the allocateAddresses
+				inNSXMacPool, err := service.isInNSXMacPool(addressBinding.MACAddress)
+				if err != nil {
+					return nil, fmt.Errorf("failed to check NSX MAC Pool: %w", err)
+				}
+				isIPPool = !inNSXMacPool
+			}
+		} else if restoreMode && len(o.Status.NetworkInterfaceConfig.IPAddresses) > 0 {
```

Comment:
For subnetport restore, I feel we should also check the mac_address, then decide allocate_addresses use IP_POOL or BOTH.

Reply:
Added, thanks!